test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/Infrastructure/DthTestServer.cs

```diff
@@ -0,0 +1,149 @@
+
+        public string Error { get; private set; }
+
+        public void WriteOutput(TextWriter writer)
+        {
+            foreach (var line in _stdout)
+            {
+                writer.WriteLine(line);
+            }
+        }
+
+        public void WriteError(TextWriter writer)
+        {
+            foreach (var line in _stderr)
+            {
+                writer.WriteLine(line);
+            }
+        }
+
+        public bool Start(TimeSpan timeout)
+        {
+            if (!File.Exists(_dthPath))
+            {
+                Error = $"Can't find {DthName} at {_dthPath}.";
+                return false;
+            }
+
+            var processStartInfo = new ProcessStartInfo()
+            {
+                UseShellExecute = false,
+                WorkingDirectory = _projectPath,
+                FileName = _bootstraperExe,
+                Arguments = $"--appbase \"{_projectPath}\" \"{_dthPath}\" {Port} {Process.GetCurrentProcess().Id} {HostId}",
+                RedirectStandardOutput = true,
+                RedirectStandardError = true
+            };
+
+            foreach (var pair in _environment)
+            {
+                processStartInfo.EnvironmentVariables[pair.Key] = pair.Value;
+            }
+
+            processStartInfo.EnvironmentVariables["DNX_TRACE"] = "1";
+
+            var wait = new ManualResetEvent(false);
+            var success = false;
+            var successOuput = $"Listening on port {Port}";
+
+            _process = Process.Start(processStartInfo);
+            _process.EnableRaisingEvents = true;
+
+            _stdout.Clear();
+
+            _process.OutputDataReceived += (sender, args) =>
+            {
+                if (args.Data != null)
+                {
+                    //Console.WriteLine(args.Data);
+                    _stdout.Add(TestUtils.RemoveAnsiColorCodes(args.Data));
+
+                    if (string.Equals(args.Data, successOuput, StringComparison.Ordinal))
+                    {
+                        wait.Set();
+                        success = true;
+                    }
+                }
+            };
+
+            _stderr.Clear();
+            _process.ErrorDataReceived += (sender, args) =>
+            {
+                if (args.Data != null)
+                {
+                    //Console.Error.WriteLine(args.Data);
+                    _stderr.Add(TestUtils.RemoveAnsiColorCodes(args.Data));
+
+                    wait.Set();
+                    success = false;
+                    Error = "Recieve data in STDERR.";
+                }
+            };
+
+            _process.BeginOutputReadLine();
+            _process.BeginErrorReadLine();
+
+            if (wait.WaitOne(timeout))
+            {
+                return success;
+            }
+            else
+            {
+                Error = $"Timeout after {timeout.Seconds}. Expected message {successOuput} from STDOUT to indicate that DTH is started successfully.";
```

Comment:
Use `timeout.TotalSeconds`. `Seconds` is just the seconds component of the date time, not the value of the timespan in seconds